go-controller/pkg/ovn/gateway.go

```diff
@@ -1093,6 +1101,38 @@ func (gw *GatewayManager) staticRouteCleanup(nextHops []net.IP) {
 	}
 }
 
+// deleteStaleJoinSubnetRoutes removes static routes referencing old join switch IP as either NextHop or IPPrefix.
+// It acts on following LRSRs:
+// 100.64.0.4               100.64.0.4 dst-ip
+func (gw *GatewayManager) deleteStaleJoinSubnetRoutes(nodeName, gwLRPIP string) error {
+	p := func(lrsr *nbdb.LogicalRouterStaticRoute) bool {
+		if lrsr.Nexthop != gwLRPIP && utilnet.IPFamilyOfString(lrsr.Nexthop) == utilnet.IPFamilyOfString(gwLRPIP) &&
+			lrsr.Nexthop == lrsr.IPPrefix {
+			networkName, isSecondaryNetwork := lrsr.ExternalIDs[types.NetworkExternalID]
+			if isSecondaryNetwork && networkName == gw.netInfo.GetNetworkName() {
+				if !config.OVNKubernetesFeature.EnableInterconnect && lrsr.ExternalIDs["nodeName"] == nodeName {
+					return true
+				} else if config.OVNKubernetesFeature.EnableInterconnect {
+					return true
+				}
+			} else if !isSecondaryNetwork {
+				if !config.OVNKubernetesFeature.EnableInterconnect && lrsr.ExternalIDs["nodeName"] == nodeName {
+					return true
+				} else if config.OVNKubernetesFeature.EnableInterconnect {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	err := libovsdbops.DeleteLogicalRouterStaticRoutesWithPredicate(gw.nbClient, gw.clusterRouterName, p)
+	if err != nil && !errors.Is(err, libovsdbclient.ErrNotFound) {
```

Comment:
I don't think that `DeleteLogicalRouterStaticRoutesWithPredicate` returns `libovsdbclient.ErrNotFound`